test/mobile/test_bytecode.py

```diff
@@ -15,33 +25,260 @@
+        '''
+
 class testVariousModelVersions(TestCase):
     def test_get_model_bytecode_version(self):
-        script_module_v4 = pytorch_test_dri / "cpp" / "jit" / "script_module_v4.ptl"
-        version_v4 = _get_model_bytecode_version(script_module_v4)
+        script_module_v4_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v4.ptl"
+        script_module_v5_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v5.ptl"
+
+        version_v4 = _get_model_bytecode_version(script_module_v4_path)
+        version_v5 = _get_model_bytecode_version(script_module_v5_path)
+
         assert(version_v4 == 4)
 
-    def test_load_and_run_model(self):
-        script_module_v4 = pytorch_test_dri / "cpp" / "jit" / "script_module_v4.ptl"
+    def test_all_backport_functions(self):
+        minimum_to_version = 4
+        class TestModule(torch.nn.Module):
+            def __init__(self, v):
+                super().__init__()
+                self.x = v
+
+            def forward(self, y: int):
+                increment = torch.ones([2, 4], dtype=torch.float64)
+                return self.x + y + increment
+
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            output_model_path = Path(tmpdirname, "script_module.ptl")
+            script_module = torch.jit.script(TestModule(1))
+            optimized_scripted_module = optimize_for_mobile(script_module)
+            exported_optimized_scripted_module = optimized_scripted_module._save_for_lite_interpreter(str(output_model_path))
+
+            # Backport model to v4 to buffer
+            output_model_path_backport = Path(tmpdirname, "script_module_backport.ptl")
+            backport_success = _backport_for_mobile(output_model_path, output_model_path_backport, 4)
+            assert(backport_success)
+
+            backport_version = _get_model_bytecode_version(output_model_path_backport)
+            assert(backport_version == 4)
+
+            shutil.rmtree(tmpdirname)
+
+
+    def test_backport_bytecode_from_file_to_file(self):
+        script_module_v5_path = pytorch_test_dri / "cpp" / "jit" / "script_module_v5.ptl"
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            backport_model_path = Path(tmpdirname, "backport_script_module_v5.ptl")
+            # backport from file
+            success = _backport_for_mobile(script_module_v5_path, backport_model_path, 4)
+            assert(success)
+
+            buf = io.StringIO()
+            torch.utils.show_pickle.main(["", tmpdirname + "/" + backport_model_path.name + "@*/bytecode.pkl"], output_stream=buf)
+            output = buf.getvalue()
+
+            expected_result = SCRIPT_MODULE_V4_path_BYTECODE_PKL
+            acutal_result_clean = "".join(output.split())
+            expect_result_clean = "".join(expected_result.split())
+            isMatch = fnmatch.fnmatch(acutal_result_clean, expect_result_clean)
+            assert(isMatch)
+
+            # Load model v5 and run forward method
```

Comment:
Comment say v5 should be v4?

Reply:
Please rename backport_script_module_v5.ptl. Hard to tell exactly what it means. I suggest: "script_module_v5_backported_to_v4.ptl" 

If this test is not meant to check explicitly for v5/v4 backporting but just the API mechanics, then I'd change the comments to reflect that. E.g.

\# Test backport API by loading model at version N and backport to version N-1
...
backport code
test
...
\# Test execution of backported N-1 model.
...